libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/fakeResolver.kt

```diff
@@ -222,31 +250,25 @@ private fun buildFieldOfNonNullType(
   }
 }
 
-class DefaultFakeResolver(val types: List<CompiledNamedType>) : FakeResolver {
-  private var currentInt = 0
-  private var currentFloat = 0.0
-  private var currentBoolean = false
-
-  private var impl = mutableMapOf<String, Int>()
+open class DefaultFakeResolver(types: List<CompiledNamedType>) : FakeResolver {
+  private val enumTypes = types.filterIsInstance<EnumType>()
+  private val allTypes = types
 
   override fun resolveLeaf(context: FakeResolverContext): Any {
     return when (val name = context.mergedField.type.rawType().name) {
-      "Int" -> currentInt++
-      "Float" -> currentFloat++
-      "Boolean" -> (!currentBoolean).also { currentBoolean = it }
+      "Int" -> context.cacheKey.hashCode() % 100
+      "Float" -> (context.cacheKey.hashCode() % 100000).toFloat() / 100.0
+      "Boolean" -> context.cacheKey.hashCode() % 2 == 0
       "String" -> {
         val index = context.path.indexOfLast { it is String }
-        context.path.subList(index, context.path.size).joinToString { it.toPathComponent() }
+        context.path.subList(index, context.path.size).joinToString(separator = "") { it.toPathComponent() }
       }
 
-      "ID" -> context.path.joinToString { it.toString() }
```

Comment:
Is this deterministic if the same object (as determined by having the same cacheKey) is reached via a different path? Probably rare that this would matter since usually an ID is going to be the cacheKey for an object, but sometimes people might have schemas that include e.g. `user { friend_user_id }` instead of `user { friend { id } }`.

Reply:
Good catch, yea, that's not deterministic. I'll fix this.